yang/frr-ospfd.yang

```diff
@@ -0,0 +1,957 @@
+        leaf advertise {
+          type boolean;
+          description
+            "Advertise this range";
+        }
+
+        leaf not-advertise {
+          type boolean;
+          description
+            "Do not advertise this range";
+        }
+
+        leaf cost {
+          type uint32 {
+            range "0..16777215";
+          }
+          description
+            "Metric for this range";
+        }
+
+        leaf substitute {
+          type inet:ipv4-prefix;
+          description
+            "Network prefix to be announced instead of range";
+        }
+      }
+    }
+  }
+
+  grouping stub-grouping {
+    container stub {
+      presence "Present when area is stub";
+      leaf no-summary {
+        type boolean;
+        description
+          "Do not inject inter-area routes into stub";
+      }
+    }
+  }
+
+  grouping shortcut-grouping {
+    container shortcut {
+      leaf default {
+        type boolean;
+        description
+          "Default shortcutting behavior";
+      }
+
+      leaf disable {
+        type boolean;
+        description
+          "Disable shortcutting through the area";
+      }
+
+      leaf enable {
+        type boolean;
+        description
+          "Enable shortcutting through the area";
+      }
+    }
+  }
+
+  grouping authentication-group {
+    container authentication {
+      presence "Enable authentication.";
+      description
+        "Enable authentication on this virtual link.";
+      leaf message-digest {
+        type boolean;
+        description
+          "Use message-digest authentication.";
+      }
+
+      leaf null {
+        type boolean;
+        description
+          "Use null authentication.";
+      }
+    }
+
+    list message-digest-key {
+      key "key-id";
+      leaf key-id {
+        type uint8;
+        description
+          "Key id";
+      }
+
+      leaf mds-key {
+        type string;
+        description
+          "The OSPF password.";
+      }
+    }
+
+    leaf authentication-key {
+      type string;
+      description
+        "The OSPF password.";
+    }
+  }
+
+  grouping virtual-link-group {
+    list virtual-link {
+      key "neighbor area-id";
+      description
+        "Virtual link parameters.";
+      leaf neighbor {
+        type inet:ipv4-address;
+        description
+          "Router ID of the remote ABR.";
+      }
+
+      leaf area-id {
+        type ospf-area-id;
+      }
+
+      uses authentication-group;
+
+      container timers {
+        leaf dead-interval {
+          type uint16;
+          units "seconds";
+          description
+            "Interval time after which a neighbor is declared down.";
+        }
+
+        leaf hello-interval {
+          type uint16 {
+            range "1..65535";
+          }
+          units "seconds";
+          description
+            "Time between HELLO packets.";
+        }
+
+        leaf retransmit-interval {
+          type uint16 {
+            range "3..65535";
```

Comment:
Why not `1..65535`, which is currently the accepted range of the "ip ospf retransmit-interval" command?

Also:
* This and several other nodes are missing default values;
* Wouldn't it be possible to create a grouping like "interface-timers" to remove the redundancy that exists between interfaces and virtual-links?